src/test_workflow/test_args.py

```diff
@@ -36,9 +38,21 @@ def __init__(self):
         self.component = args.component
         self.keep = args.keep
         self.logging_level = args.logging_level
-        self.path = args.path if validators.url(args.path) else os.path.realpath(args.path)
+        self.test_manifest_path = self._validate_path(args.test_manifest_path)
+
+        self.opensearch_path = self._validate_paths(args.paths, "opensearch")
+        self.opensearch_dashboards_path = self._validate_paths(args.paths, "opensearch-dashboards")
+
+    def _validate_paths(self, args_paths: dict, name: str) -> str:
```

Comment:
`_validate_paths` is really an existing getter with default: `self.paths.get('opensearch-dashboards', '.')`, I would remove that method

there's a way to move the call of `_validate_path` to the `parser` as well, but I forget how to do that, NBD

Reply:
will have follow ups on this.